parts/windowsProvision.ps1

```diff
@@ -50,115 +298,137 @@ function InstallOpenSSH()
+function Install-Ocaml {
+    $installDir = Join-Path $env:ProgramFiles "ocpwin64"
+    try {
+        Install-ZipCITool -ZipPath $PACKAGES["ocaml"]["local_file"] `
+                          -InstallDirectory $installDir `
+                          -EnvironmentPath @("$installDir\4.02.1+ocp1-msvc64-20160113\bin")
+    } catch {
+        Remove-Item -Recurse -Force $installDir
+        Throw
+    }
+    pushd "${env:ProgramFiles}\ocpwin64\4.02.1+ocp1-msvc64-20160113\bin"
+    & ./ocpwin -in
+}
 
-sleep 5
 
 try {
-   Start-Service "AESMService"
-}
-catch {
-    Write-Output "Could not start service: $_"
-}
-
-# Download useful tools to C:\Bin.
-Write-Output "get nuget"
-& curl.exe https://dist.nuget.org/win-x86-commandline/v4.1.0/nuget.exe -o C:\bin\nuget.exe
-
-# Download the Build Tools bootstrapper outside of the PATH.
-Write-Output "get visual stdio"
-& curl.exe -L -o "C:\TMP\vs_buildtools.exe"  "https://aka.ms/vs/15/release/vs_buildtools.exe" 
-
-Write-Output "install visual stdio"
-$argslist = " -q --wait --norestart --nocache "
-$argslist += "    --add Microsoft.VisualStudio.Workload.MSBuildTools "
-$argslist += "    --add Microsoft.VisualStudio.Workload.VCTools "
-$argslist += "    --add Microsoft.VisualStudio.Component.VC.Tools.x86.x64 "
-$argslist += "    --add Microsoft.VisualStudio.Component.VC.140 "
-$argslist += "    --add Microsoft.VisualStudio.Component.Windows10SDK.16299.Desktop "
-$argslist += "    --add Microsoft.VisualStudio.Component.Windows81SDK "
-$argslist += "    --add Microsoft.VisualStudio.Component.VC.ATL "
-
-Start-Process -Wait -FilePath "C:\TMP\vs_buildtools.exe" -ArgumentList $argslist
-
-$env:PATH += ";c:/program files/7-zip"
-
-Write-Output "get cmake"
-$cmake_uri = "https://cmake.org/files/v3.13/cmake-3.13.0-rc1-win64-x64.msi"
-$cmake_file = "c:/tmp/cmake-3.13.0-rc1-win64-x64.msi"
-& curl.exe -L -o $cmake_file $cmake_uri
-Start-Process -Wait -FilePath $cmake_file -ArgumentList " /quiet /passive"
-
-#
-# ocaml for building oeedgr8r
-#
-Write-Output "get ocaml"
-$ocaml_file = "c:/tmp/ocpwin64.zip"
-$ocaml_install_dir = "c:/Program Files/ocpwin64"
-$ocaml_uri  = "http://www.ocamlpro.com/pub/ocpwin/ocpwin-builds/ocpwin64/20160113/ocpwin64-20160113-4.02.1+ocp1-msvc64.zip"
-& curl.exe -o $ocaml_file $ocaml_uri
-& 7z x $ocaml_file -o"c:/Program Files/ocpwin64"
-pushd "C:\Program Files\ocpwin64\4.02.1+ocp1-msvc64-20160113\bin"
-& ./ocpwin -in
+    New-Directory -Path ${AZUREDATA_DIRECTORY}
+    New-Directory -Path ${AZUREDATA_BIN_DIRECTORY}
+
+    Install-Git
+    Install-7Zip
+    Install-OpenSSH
+
+    $is_vanilla = "IS_VANILLA_VM"
+    if ($is_vanilla -eq "true") {
+        Write-Output "Skipping Open Enclave installation."
+        exit 0
+    }
+    Write-Output "Installing Open Enclave"
+
+    Install-SGX
+    Install-PSW
+
+    sleep 5
+
+    try {
+       Start-Service "AESMService"
+    }
+    catch {
+        Write-Output "Could not start service: $_"

```

Comment:
I don't think we want to ignore any service start failure for the `AESMService`.